activerecord/lib/active_record/tasks/sqlite_database_tasks.rb

```diff
@@ -33,9 +33,11 @@ def drop
 
       def purge
         drop
+        connection.disconnect!
       rescue NoDatabaseError
       ensure
         create
+        connection.reconnect!
```

Comment:
Action Mailbox and other related libraries are relying on loading the schema before test runs via `db:test:prepare` which calls `db:schema:load` which calls `db:purge`. With the changes in connection management we no longer are re-establishing the connection so anything that called purge would fail with a sqlite3 error. I noticed that this behavior was not reproducible in mysql2 and found that the `purge` method there reconnects to the new database. I implemented that same behavior in `purge` here. Previously this was working "fine" because we were always removing the connection even if it was the same. Now that we're not removing the connection we want to reset it so we don't have a bad connection.